doc/developer/design/20250226_postgres_style_explain.md

```diff
@@ -0,0 +1,338 @@
+syntax (fixing [a bug with `Let` and `LetRec`
+rendering](https://github.com/MaterializeInc/database-issues/issues/8993)
+in the process.
+
+We will need three pieces of work, which should all land together:
+
+ - We must implement the new output and put in appropriate SLT tests
+   for it.
+ - We must update `mz_lir_mapping` to use the new vocabulary.
+ - We must update the documentation to explain the new output, ideally
+   using this output everywhere `EXPLAIN` is used.
+
+### Concrete Mapping
+
+| LIR node    | `mz_lir_mapping` node                    | New, Postgres-style syntax              |
+| :---------- | :--------------------------------------- | :-------------------------------------- |
+| `Constant`  | `Constant`                               | `Constant`                              |
+| `Get`       | `Get::PassArrangements l0`               | `Index Scan on l0 using ...`            |
+| `Get`       | `Get::Arrangement l0 (val=...)`          | `Index Lookup on l0 using ...`          |
+| `Get`       | `Get::Collection l0`                     | `Read l0`                               |
+| `Mfp`       | `MapFilterProject`                       | `Map/Filter/Project`                    |
+| `FlatMap`   | `FlatMap`                                | `Flat Map`                              |
+| `Join`      | `Join::Differential`                     | `Differential Join`                     |
+| `Join`      | `Join::Delta`                            | `Delta Join`                            |
+| `Reduce`    | `Reduce::Distinct`                       | `Distinct GroupAggregate`               |
+| `Reduce`    | `Reduce::Accumulable`                    | `Accumulable GroupAggregate`            |
+| `Reduce`    | `Reduce::Hierarchical (monotonic)`       | `Monotonic Hierarchical GroupAggregate` |
+| `Reduce`    | `Reduce::Hierarchical (buckets: ...)`    | `Bucketed Hierarchical GroupAggregate`  |
+| `Reduce`    | `Reduce::Basic`                          | `Non-incremental GroupAggregate`        |
+| `Reduce`    | `Reduce::Collation`                      | `Collated GroupAggregate` (details?)    |
+| `TopK`      | `TopK::MonotonicTop1`                    | `Monotonic Top1`                        |
+| `TopK`      | `TopK::MonotonicTopK`                    | `Monotonic TopK`                        |
+| `TopK`      | `TopK::Basic`                            | `Non-monotonic TopK`                    |
+| `Negate`    | `Negate`                                 | `Negate Diffs`                          |
+| `Threshold` | `Threshold`                              | `Threshold Diffs`                       |
+| `Union`     | `Union`                                  | `Union`                                 |
+| `Union`     | `Union (consolidates output)`            | `Consolidating Union`                   |
+| `ArrangeBy` | `Arrange`                                | `Arrange`                               |
```

Comment:
If there are no keys, but `raw` is true, then maybe think up another name for this case.

Reply:
Yes, good idea! Feels related to the `Get` refactoring.

Reply:
I propose we write `Stream/Arrange` when `raw=true`, on the principle that we should indicate when we _could_ be streaming rather than using an arrangement.

Reply:
- If there are no `keys`, but there is `raw`, then it could be something like `Stream` or `Unarrange`.
- If it's just `keys` but no `raw`, then `Arrange` is fine.

But I'm not sure what to do with the weird cases:
- Both `keys` and `raw`: how to indicate that there are two things being produced? Or maybe it's enough to just say `Arrange`, because the `raw` part won't be important performance-wise in this case?
- Neither `keys` nor `raw`: hopefully this won't occur, because it doesn't make a whole lot of sense :D